tests/testthat/test-toolFixWeight.R

```diff
@@ -0,0 +1,71 @@
+test_that("toolFixWeight works with one dimensional weight", {
+  map <- data.frame(from = c("A", "A", "B", "B"),
+                    to = c("A1", "A2", "B1", "B2"))
+  weight <- new.magpie(map$to, fill = 0)
+  weight["B1", , ] <- 1
+
+  fixedWeight <- toolFixWeight(weight, map, dim = 1)
+  expected <- new.magpie(c("A1", "A2", "B1", "B2"))
+  expected[c("A1", "A2"), , ] <- 10^-30
+  expected["B1", , ] <- 1
+  expected["B2", , ] <- 0
+  expect_identical(fixedWeight, expected)
+})
+
+test_that("toolFixWeight works with three dimensional weight", {
+  map <- data.frame(from = c("CC", "CC", "CC", "DD", "DD"),
+                    to = c("CC1", "CC2", "CC3", "DD1", "DD2"))
+  weight <- new.magpie(c("A", "B"), map$to, c("EEE", "FFF"), fill = 0)
+  weight["B", "CC3", "EEE"] <- 1
+
+  fixedWeight <- toolFixWeight(weight, map, dim = 2)
+  expected <- weight
+  expected[] <- 10^-30
+  expected["B", "CC3", "EEE"] <- 1
+  expected["B", c("CC1", "CC2"), "EEE"] <- 0
+  expect_identical(fixedWeight, expected)
+})
+
+test_that("toolFixWeight works with subdims", {
+  map <- data.frame(from = c("CC", "CC", "CC", "DD", "DD"),
+                    to = c("CC1", "CC2", "CC3", "DD1", "DD2"))
+  weight <- new.magpie(c("A", "B"), paste0("X.", map$to), c("EEE", "FFF"), fill = 0)
+  weight["B", "CC3", "EEE"] <- 1
+
+  fixedWeight <- toolFixWeight(weight, map, dim = 2.2)
+  expected <- weight
+  expected[] <- 10^-30
+  expected["B", "CC3", "EEE"] <- 1
+  expected["B", c("CC1", "CC2"), "EEE"] <- 0
+  expect_identical(fixedWeight, expected)
+})
+
+test_that("toolFixWeight works with multi subdims map", {
+  map <- data.frame(from = c("X.CC", "X.CC", "X.CC", "X.DD", "X.DD"),
+                    to = c("X.CC1", "X.CC2", "X.CC3", "X.DD1", "X.DD2"))
+  weight <- new.magpie(c("A", "B"), map$to, c("EEE", "FFF"), fill = 0)
+  weight["B", "CC3", "EEE"] <- 1
+
+  fixedWeight <- toolFixWeight(weight, map, dim = 2)
+  expected <- weight
+  expected[] <- 10^-30
+  expected["B", "CC3", "EEE"] <- 1
+  expected["B", c("CC1", "CC2"), "EEE"] <- 0
+  expect_identical(fixedWeight, expected)
+})
+
+test_that("toolFixWeight can handle large objects", {
+  skip("to save time skip test whether toolFixWeight can handle large objects")
```

Comment:
hm... I am wondering whether we want to have that conditional on the context, e.g. only skip locally but run on CI (although running performance tests on CI is somewhat flaky by itself :|). As I can not think of the context in which running this is perfectly fine, maybe we need to keep it skipped...